### ISSUE TITLE
Add support for custom commands on motion detected.

### ISSUE THREAD

**Issue by rayzorben** [bluecherrydvr/bluecherry-apps#373]
Would like to call out to my own service/webhook/application when motion is detected. This could be things like
* IFTTT
* Pushover
* Home Assistant

For example, my main use is Pushover as it seems the fastest in sending me alerts when motion is detected. IFTTT is 5-10 seconds, Email can be 1 second or 20 seconds. Pushover when called directly seems to be under 1 second. https://support.pushover.app/i44-example-code-and-pushover-libraries

However, it would be nice to support it in a generic way, whether that is calling out via shell exec to a command line application or? 

It would support a set of variables such as
`Device` e.g., Front Porch
`Username` e.g. rayzorben
`OccuredAt` e.g. Wed Dec 04 4:34PM
`Server` e.g. Home
`ImageFilename` e.g. /tmp/2019120316340033717.jpg which is an image on disk that represents the motion.

**Comment by curtishall** [comment]
Similar to issue #314 something that we need to add in version 3 for sure.

It won't make 3.0 as we already have that in production but would be something to add in later versions.

**Comment by curtishall** [comment]
Thinking aloud you might be able to add onto the email notification routine and upload images and notifications that way in the short term - https://support.pushover.app/i44-example-code-and-pushover-libraries#unix-image

**Comment by rayzorben** [comment]
> Thinking aloud you might be able to add onto the email notification routine and upload images and notifications that way in the short term - https://support.pushover.app/i44-example-code-and-pushover-libraries#unix-image

Yep! Right now I am using the email capabilities of Pushover, but it still has to go through gmail SMTP server. And for some reason sometimes I get those in a heartbeat, and another times it just lags for 10-15 seconds. Right now it is 'good enough' but if I get frustrated I will just do what you suggest and modify the code around emails to call into libcurl or something using that API.

**Comment by maiksicks** [comment]
We can easily customize the file `/usr/share/bluecherry/www/lib/mailer.php` and add following line of code (shell_exec):
 
```php
if ($argv[1] == "motion_event") {
     
    ...
    
    #run shell script
    shell_exec("/bin/sh /usr/share/bluecherry/events/motion.sh {$device_id} {$timestamp}");
}
```

This executes a shell script on every motion. @rayzorben You also could execute your pushover command directly (as HTTP request over cURL). It is possibly not the best "solution" but it works fine until this feature is built-in.
